src/App.js

```diff
@@ -11,34 +13,181 @@ class App extends React.Component {
+      let currentGamesWon = this.state.gamesWon.slice(0);
+      let updatedGamesWon = this.computeGamesWon(decision, currentGamesWon);
+      console.log("computeOutcome() gamesWon variable:" + updatedGamesWon);
+
+      return {
+        // Remove last 2 cards from cardDeck
+        cardDeck: [],
+        // Deal last 2 cards to currCards
+        currCards: [],
+        score: newScore,
+        winner: decision,
+        gameEnded: true,
+        gamesWon: updatedGamesWon,
+      };
+    });
+  };
+
   dealCards = () => {
-    this.setState((state) => ({
-      // Remove last 2 cards from cardDeck
-      cardDeck: state.cardDeck.slice(0, -2),
-      // Deal last 2 cards to currCards
-      currCards: state.cardDeck.slice(-2),
-    }));
+    this.setState((state) => {
+      let cards = state.cardDeck.slice(-2);
+      let currentScore = this.state.score.slice(0);
+      let newScore = this.computeTurnWinner(cards, currentScore);
+      console.log(`dealCards() state variable gamesWon: ${state.gamesWon}`);
+
+      return {
+        // Remove last 2 cards from cardDeck
+        cardDeck: this.state.roundsLeft > 1 ? state.cardDeck.slice(0, -2) : [],
+        // Deal last 2 cards to currCards
+        currCards: state.cardDeck.slice(-2),
+        score: newScore,
+        roundsLeft: state.roundsLeft - 1,
+      };
+    });
+
+    console.log(`card deck length: ${this.state.cardDeck.length}`);
   };
 
-  render() {
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
-    ));
+  decideWinner(currentScore) {
+    let decision = "";
+    if (currentScore[0] > currentScore[1]) {
+      decision = "Player 1 won";
+    } else if (currentScore[0] < currentScore[1]) {
+      decision = "Player 2 won";
+    } else if (currentScore[0] === currentScore[1]) {
+      decision = "It's a tie";
+    }
+    return decision;
+  }
 
+  computeTurnWinner(cards, currentScore) {
+    if (cards[0].rank === cards[1].rank) {
+      currentScore[0]++;
+      currentScore[1]++;
+    } else if (cards[0].rank > cards[1].rank) {
+      currentScore[0]++;
+    } else {
+      currentScore[1]++;
+    }
+    return currentScore;
+  }
+
+  computeGamesWon(winner, gamesWon) {
+    if (winner === "Player 1 won") {
+      gamesWon[0]++;
+    } else if (winner === "Player 2 won") {
+      gamesWon[1]++;
+    }
+    return gamesWon;
+  }
+
+  render() {
     return (
       <div className="App">
         <header className="App-header">
-          <h3>High Card 🚀</h3>
-          {currCardElems}
+          <h1>High Card 🚀</h1>
+
+          <PlayingCards currCards={this.state.currCards} />
+          {this.state.gameEnded ? this.state.winner : ""}
           <br />
-          <button onClick={this.dealCards}>Deal</button>
+          <button
+            onClick={
+              this.state.gameEnded
```

Comment:
i strongly discourage using nested ternaries like this because it makes the code a lot harder to read.

it also encourages the next developer who's working on your nested ternary to add more ternaries (making the problem worse) when they have to add conditions to your existing code

Reply:
you could do something like:

```javascript
function handleButtonClick() {
  if (this.state.gameEnded) this.resetGame(); // don't forget the brackets to invoke the function!
  else if (this.state.roundsLeft) this.dealCards();
  else this.computeOutcome();
}
```